operator/v1/types_ingress.go

```diff
@@ -225,6 +232,132 @@ var (
 	DNSReadyIngressConditionType = "DNSReady"
 )
 
+// SecuritySpec defines the settings for securing IngressController connections.
+type SecuritySpec struct {
+	// Profile is one of "Old", "Intermediate", "Modern" or "Custom". "Old",
+	// "Intermediate" and "Modern" profiles map to security configurations from [1]:
+	//
+	// [1] https://wiki.mozilla.org/Security/Server_Side_TLS#Recommended_configurations
+	//
+	// When a profile of type "Old", "Intermediate" or "Modern" is set, the CustomSettings
+	// field is ignored.
```

Comment:
or forbidden? The union type proposal in upstream requires that `customSetting` is undefined if another type is set.

Reply:
@sttts thank you for the review. I will change undefined to forbidden.